ocrd_network/ocrd_network/processing_server.py

```diff
@@ -670,3 +687,41 @@ async def list_processors(self) -> List[str]:
             unique_only=True
         )
         return processor_names_list
+
+    # TODO: think about providing arguments in another way
+    # TODO: this function "just" writes to the queue and returns. A network-client functionality
+    #       should be available to react to everys processors callback. With this feedback
+    #       a blocking mechanism could be provided to inform about starting the cain and waiting for
+    #       the processors to finish and printing when reponses are received from the processors
```

Comment:
Indeed, that is a tricky but important aspect- how can we manage the callbacks in such a way that it is useful to the user? We could calculate the number of jobs we'll create (i.e. the output of this to-be-refactored `task_sequence_to_processing_jobs` function I mention below) and send back the current state, i.e. `Job {uuid} [30 of 123] finished {"successful" if state == 'SUCCESS` else "with an eror"}.

But more importantly, could we design this in such a way that we only give a callback if a) one job has failed or b) the complete workflow succeeded? Can we perhaps add a "session" ID to the jobs which the processing server can track and call the callback only for the aforementioned cases?

Reply:
I am not sure I fully understand the idea. The `callback_url` passed to the `run_workflow` method is there to be called only once - when the entire workflow succeeds or fails.

Reply:
Oh, okay, I thought the `callback_url` was to be passed on to every `PYInputJob` and I expected it to be called when every **job** is finished, not just if the `workflow` finished. If that is the case already, then disregard my comment :)

Reply:
Maybe we should think about what response to return instead of returning just a list of `PYJobOutput`s to make it more useful for the user. The `callback_url` concept was planned to be used by the `Workflow Server` to report the workflow status. Not sure how this should be used to make it useful to the user as well.